Firestore/core/src/firebase/firestore/api/document_reference.h

```diff
@@ -66,26 +71,29 @@ class DocumentReference {
 
   const std::string& document_id() const;
 
-  FIRCollectionReference* Parent() const;
+  // TODO(varconst) uncomment when core API CollectionReference is implemented.
+  // CollectionReference Parent() const;
 
   std::string Path() const;
 
-  FIRCollectionReference* GetCollectionReference(
-      const std::string& collection_path) const;
+  // TODO(varconst) uncomment when core API CollectionReference is implemented.
```

Comment:
Because the core `DocumentReference` no longer has a `FIRFirestore` pointer, and `FIRFirestore` cannot be directly created from just a core `Firestore`, it can only create a core `CollectionReference`, which isn't implemented yet, not `FIRCollectionReference`.

Reply:
My proposed changes in #2507 make this possible.

Reply:
Cool. I'd still prefer to do this in a follow-up, to minimize the size of this PR.

Reply:
Yep, no problem.